assets/js/modules/analytics/components/dashboard/DashboardUniqueVisitorsWidget.js

```diff
@@ -0,0 +1,151 @@
+import { STORE_NAME as CORE_SITE } from '../../../../googlesitekit/datastore/site/constants';
+import { STORE_NAME as CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import whenActive from '../../../../util/when-active';
+import PreviewBlock from '../../../../components/preview-block';
+import DataBlock from '../../../../components/data-block';
+import Sparkline from '../../../../components/sparkline';
+import AnalyticsInactiveCTA from '../../../../components/analytics-inactive-cta';
+import { extractAnalyticsDashboardSparklineData, parseTotalUsersData } from '../../util';
+import { extractForSparkline, getSiteKitAdminURL, changeToPercent, readableLargeNumber } from '../../../../util';
+import getDataErrorComponent from '../../../../components/notifications/data-error';
+import getNoDataComponent from '../../../../components/notifications/nodata';
+
+const { useSelect } = Data;
+
+function DashboardUniqueVisitorsWidget() {
+	const {
+		loading,
+		error,
+		sparkData,
+		visitorsData,
+	} = useSelect( ( select ) => {
+		const store = select( STORE_NAME );
+		const args = {
+			dateRange: select( CORE_USER ).getDateRange(),
+		};
+
+		const url = select( CORE_SITE ).getCurrentEntityURL();
+		if ( url ) {
+			args.url = url;
+		}
+
+		const sparkDataArgs = {
+			compareDateRanges: 1,
+			dimensions: 'ga:date',
+			metrics: [
+				{
+					expression: 'ga:users',
+					alias: 'Users',
+				},
+				{
+					expression: 'ga:sessions',
+					alias: 'Sessions',
+				},
+				{
+					expression: 'ga:bounceRate',
+					alias: 'Bounce Rate',
+				},
+				{
+					expression: 'ga:avgSessionDuration',
+					alias: 'Average Session Duration',
+				},
+				{
+					expression: 'ga:goalCompletionsAll',
+					alias: 'Goal Completions',
+				},
+			],
+			limit: 180,
+			...args,
+		};
+
+		const visitorsDataArgs = {
+			multiDateRange: 1,
+			metrics: [
+				{
+					expression: 'ga:users',
+					alias: 'Total Users',
+				},
+			],
+			...args,
+		};
+
+		return {
+			loading: store.isResolving( 'getReport', [ sparkDataArgs ] ) || store.isResolving( 'getReport', [ visitorsDataArgs ] ),
+			error: store.getErrorForSelector( 'getReport', [ sparkDataArgs ] ) || store.getErrorForSelector( 'getReport', [ visitorsDataArgs ] ),
+			// Due to the nature of these queries, we need to run them separately.
+			sparkData: store.getReport( sparkDataArgs ),
+			visitorsData: store.getReport( visitorsDataArgs ),
+		};
+	} );
+
+	if ( loading ) {
+		return <PreviewBlock width="100%" height="202px" />;
+	}
+
+	if ( error ) {
+		return getDataErrorComponent( __( 'Analytics', 'google-site-kit' ), error.message );
+	}
+
+	if ( ( ! sparkData || ! sparkData.length ) && ( ! visitorsData || ! visitorsData.length ) ) {
+		return getNoDataComponent( __( 'Analytics', 'google-site-kit' ) );
+	}
+
+	const extractedAnalytics = extractAnalyticsDashboardSparklineData( sparkData );
+
+	const { previousTotalUsers, totalUsers } = parseTotalUsersData( visitorsData );
```

Comment:
Let's not use this function and instead extract the values manually like it is done in the Bounce Rate widget above.